src/mjolnir/ferry_connections.cc

```diff
@@ -61,7 +61,9 @@ uint32_t ShortestPath(const uint32_t start_node_idx,
   uint16_t overall_access_before = baldr::kVehicularAccess;
   uint16_t overall_access_after = 0;
   // find accessible paths for as many modes as we can but stop if we can't find any more
-  while (overall_access_before != overall_access_after) {
+  // or found all modes in the first round
+  while ((overall_access_after != baldr::kVehicularAccess) &&
+         (overall_access_before != overall_access_after)) {
```

Comment:
tiny optimization: if we actually found all modes in the first expansion, we can stop. before it would do another round, only to find the exact same path.